frontend/src/components/Form.js

```diff
@@ -0,0 +1,171 @@
+      body: JSON.stringify({ email, password }),
+      headers: { 'Content-Type': 'application/json' }
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw 'Login failed'
+        }
+        return res.json()
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFail(err))
+  }
+
+  const handleSignup = (event) => {
+    event.preventDefault()
+
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, email, password }),
+      headers: { 'Content-Type': 'application/json' }
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw 'Signup failed'
+        }
+        return res.json()
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFail(err))
+  }
+
+  const handleNewUser = (event) => {
+    event.preventDefault()
+    setIsNewUser(true)
+    setEmail('')
+    setPassword('')
+  }
+  if (accessToken) {
+    return <Secrets />
+  }
+  if (!isNewUser) {
+    return (
+      <>
+        <FormStyle onSubmit={handleLogin}>
+          <Title>Login</Title>
+          <Input
+            type="email"
+            value={email}
+            placeholder="email"
+            onChange={(event) => setEmail(event.target.value)}
+            required />
+          <Input
+            type="password"
+            value={password}
+            placeholder="password"
+            onChange={(event) => setPassword(event.target.value)}
+            required />
+          <Button
+            type="submit">
+            Login
+          </Button>
+          {loginFailed && <Paragraph>Wrong email or password. Please try again.</Paragraph>}
+          <Button
+            type="submit"
+            onClick={handleNewUser}>
+            Sign up
+          </Button>
+        </FormStyle>
+      </>
+    )
+  }
+
+  return (
+    <FormStyle onSubmit={handleSignup}>
+      <Title>Signup</Title>
+      <Input
+        type="email"
+        value={email}
+        placeholder="email"
+        onChange={(event) => setEmail(event.target.value)}
+        required />
+      <Input
+        type="text"
+        value={name}
+        placeholder="username"
+        onChange={(event) => setName(event.target.value)}
+        minLength="2"
+        maxLength="20"
+        required />
+      <Input
+        type="password"
+        value={password}
+        placeholder="password"
+        onChange={(event) => setPassword(event.target.value)}
+        required />
+      <Input
+        type="password"
+        value={confirmedPassword}
+        placeholder="retype password"
+        onChange={(event) => setConfirmedPassword(event.target.value)}
+        required />
+      <Button
+        type="submit"
+        disabled={password !== confirmedPassword || password.length === 0}>
+        Sign up
+      </Button>
+      {loginFailed && <Paragraph>Could not create user. Please try again.</Paragraph>}
+    </FormStyle>
+  )
+}
```

Comment:
It is a very bad practice to have two separate returns in one component. You should create one form with two modes (in/up) and use React Router to open the form in sign-up or sign-in mode passed by parameter. 🤓